backend/server/src/models/error.rs

```diff
@@ -3,31 +3,42 @@ use axum::response::{IntoResponse, Redirect, Response};
 
 /// Custom error enum for Chaos.
 ///
-/// Handles all anyhow errors (when `?` is used) alongside
+/// Handles all errors thrown by libraries (when `?` is used) alongside
 /// specific errors for business logic.
+#[derive(thiserror::Error, Debug)]
 pub enum ChaosError {
-    NotLoggedIn,
-    Unauthorized,
-    ForbiddenOperation,
-    ServerError(anyhow::Error),
+    #[error("Not logged in")]
+    NotLoggedInError,
+
+    #[error("Not authorized")]
+    UnauthorizedError,
+
+    #[error("Forbidden operation")]
+    ForbiddenOperationError,
```

Comment:
Just personal opinion/taste: I like how you named error enum variant that wrap actual error types (database, reqwest, oath) with an `Error` suffix, but I would personally just leave out the `Error` suffix for the others (not logged in, unauthorized, forbidden), as they will always be used as `ChaosError::<Variant>` anyways, which makes it clear that it's an error. Like I said, this is just personal preference, so do with it as you wish.